src/Servers/Connections.Abstractions/src/ConnectionContext.cs

```diff
@@ -12,22 +12,50 @@
 
 namespace Microsoft.AspNetCore.Connections
 {
+    /// <summary>
+    /// Encapsulates all information about an individual connection irrespective of the underlying transport.
+    /// </summary>
     public abstract class ConnectionContext : IAsyncDisposable
     {
+        /// <summary>
+        /// Gets or sets a unique identifier to represent this connection in trace logs.
+        /// </summary>
         public abstract string ConnectionId { get; set; }
 
+        /// <summary>
+        /// Gets the collection of features provided by the server and middleware available on this connection.
+        /// </summary>
         public abstract IFeatureCollection Features { get; }
 
+        /// <summary>
+        /// Gets or sets a key/value collection that can be used to share data within the scope of this connection.
+        /// </summary>
         public abstract IDictionary<object, object> Items { get; set; }
 
+        /// <summary>
+        /// Gets or sets the <see cref="IDuplexPipe"/> that can be used to read and write to the this connection.
+        /// </summary>
         public abstract IDuplexPipe Transport { get; set; }
 
+        /// <summary>
+        /// Notifies when the client connection has closed.
```

Comment:
```suggestion
        /// Triggered when the client connection is closed.
```

(Source for precedent: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.iapplicationlifetime?view=aspnetcore-2.2)